### ISSUE TITLE
[2.0.3] java.lang.invoke.BoundMethodHandle$Species_LL cannot be cast to java.lang.invoke.SimpleMethodHandle

### ISSUE THREAD

**Issue by senzu** [quarkusio/quarkus#19019]
### Describe the bug

A simple toString() function throws ClassCastException

java.lang.ClassCastException: java.lang.invoke.BoundMethodHandle$Species_LL cannot be cast to java.lang.invoke.SimpleMethodHandle
        at com.oracle.svm.methodhandles.MethodHandleIntrinsicImpl.execute(MethodHandleIntrinsicImpl.java:258)
        at com.oracle.svm.methodhandles.Util_java_lang_invoke_MethodHandle.invokeInternal(Target_java_lang_invoke_MethodHandle.java:144)
        at java.lang.invoke.MethodHandle.invokeBasic(MethodHandle.java:79)
        at java.lang.invoke.LambdaForm$NamedFunction.invokeWithArguments(LambdaForm.java:76)
        at java.lang.invoke.LambdaForm.interpretName(LambdaForm.java:981)
        at java.lang.invoke.LambdaForm.interpretWithArguments(LambdaForm.java:958)
        at java.lang.invoke.MethodHandle.invokeBasic(MethodHandle.java:85)
        at java.lang.invoke.MethodHandle.invokeBasic(MethodHandle.java:0)
        at java.lang.invoke.LambdaForm$MH/373406506.invoke(LambdaForm$MH)
        at hu.tsystems.database.entity.MeClapSzam.toString(MeClapSzam.java:431)

**Comment by geoand** [comment]
I assume this is happening in native mode, correct?

What Quarkus versions and what GraalVM version are you using?

**Comment by senzu** [comment]
Yes, it happens in native mode!

The GraalVM version is the following:

openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05, mixed mode, sharing)

But I have built it with Docker native command:

mvn clean package -Pnative -Dquarkus.native.container-build=true -Dquarkus.container-image.build=true

**Comment by geoand** [comment]
cc @zakkak @galderz

**Comment by galderz** [comment]
The local GraalVM version has no effect if building in container, so what matters here is what Quarkus version you're using (???) and see what container build image is in use, and see what GraalVM / Mandrel version is used then...

**Comment by galderz** [comment]
Hmmm, the default container builder image in 2.0.3.Final appears to use GraalVM 21.1 (see [here](https://github.com/quarkusio/quarkus/blob/2.0.3.Final/build-parent/pom.xml#L134)). If this is still happening in 21.1.x, could you share a reproducer?

**Comment by galderz** [comment]
One more detail, the issue was only fixed in 21.2 (see [fix](https://github.com/oracle/graal/commit/540eff5e0b6edf7cffafba9fad5c874446d5299e)).

**Comment by geoand** [comment]
@senzu can you try with GraalVM 21.2?

**Comment by senzu** [comment]
Hi guys? How can I try GraalVM 21.2? I have upgraded the Quarkus version to 2.1.1.Final, but the container builder image is still 21.1.0

[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM 21.1.0 Java 11 CE (Java Version 11.0.11+8-jvmci-21.1-b05)

Unfortunately I'm on Windows, and if I create the native image with GraalVM an error happens, so I have to use Docker

**Comment by zakkak** [comment]
> Hi guys. How can I try GraalVM 21.2? I have upgraded the Quarkus version to 2.1.1.Final, but the container builder image is still 21.1.0
> 
> [INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM 21.1.0 Java 11 CE (Java Version 11.0.11+8-jvmci-21.1-b05)
> 
> Unfortunately I'm on Windows, and if I create the native image with GraalVM an error happens, so I have to use Docker

Hi @senzu, please try building with:
```
mvn clean package -Pnative -Dquarkus.native.container-build=true \
    -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:21.2-java11 \
    -Dquarkus.container-image.build=true
```

**Comment by senzu** [comment]
> > Hi guys. How can I try GraalVM 21.2? I have upgraded the Quarkus version to 2.1.1.Final, but the container builder image is still 21.1.0
> 
> Hi @senzu, please try building with:
> 
> ```
> mvn clean package -Pnative -Dquarkus.native.container-build=true \
>     -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:21.2-java11 \
>     -Dquarkus.container-image.build=true
> ```

Hi zakkak, it works without error with ubi-quarkus-native-image:21.2-java11!

**Comment by zakkak** [comment]
> Hi @zakkak, it works without error with ubi-quarkus-native-image:21.2-java11!

Nice, 21.2 will become the default with Quarkus 2.2 (https://github.com/quarkusio/quarkus/pull/18932) and it will be (re)considered for backporting to Quarkus 2.1 (there were some issues with that).

Thanks for testing!

**Comment by senzu** [comment]
Issue solved in GraalVM 21.2, which will be become default in Quarkus 2.2
